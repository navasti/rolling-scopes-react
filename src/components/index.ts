export { Loader } from './Loader';
export { Modal } from './Modal';
