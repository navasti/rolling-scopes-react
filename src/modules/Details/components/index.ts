export { MoveCardDetails } from './MoveCardDetails';
export { TypeCardDetails } from './TypeCardDetails';
export { PokemonCardDetails } from './PokemonCardDetails';
