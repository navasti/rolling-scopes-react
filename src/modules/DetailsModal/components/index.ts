export { TypeModalContent } from './TypeModalContent';
export { MoveModalContent } from './MoveModalContent';
export { PokemonModalContent } from './PokemonModalContent';
