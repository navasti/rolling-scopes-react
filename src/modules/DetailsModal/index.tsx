import { PokemonModalContent, MoveModalContent, TypeModalContent } from './components';
import { PokemonDetails, PokemonMoveDetails, PokemonTypeDetails } from 'types';
import { Modal } from 'components';
import { forwardRef } from 'react';
import * as S from './styled';

type Props = {
  selectedType?: PokemonTypeDetails;
  selectedMove?: PokemonMoveDetails;
  selectedPokemon?: PokemonDetails;
  handleCloseModal: () => void;
  isModalOpened: boolean;
};

export const DetailsModal = forwardRef<HTMLDivElement, Props>((props, ref) => {
  const { handleCloseModal, isModalOpened, selectedMove, selectedPokemon, selectedType } = props;
  return (
    <>
      <Modal handleCloseModal={handleCloseModal} opened={isModalOpened} ref={ref}>
        <S.ModalTitle>
          {selectedPokemon
            ? selectedPokemon.name
            : selectedMove
            ? selectedMove.name
            : selectedType
            ? selectedType.name
            : null}
        </S.ModalTitle>
        {selectedPokemon ? (
          <PokemonModalContent selectedPokemon={selectedPokemon} />
        ) : selectedMove ? (
          <MoveModalContent selectedMove={selectedMove} />
        ) : selectedType ? (
          <TypeModalContent selectedType={selectedType} />
        ) : (
          <div></div>
        )}
      </Modal>
    </>
  );
});
