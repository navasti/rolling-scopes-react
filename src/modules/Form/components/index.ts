export { BirthdayField } from './BirthdateField';
export { ErrorMessage } from './ErrorMessage';
export { ConsentField } from './ConsentField';
export { AvatarField } from './AvatarField';
export { GenderField } from './GenderField';
export { ShinyField } from './ShinyField';
export { TypeField } from './TypeField';
export { NameField } from './NameField';
export { FormCard } from './FormCard';
