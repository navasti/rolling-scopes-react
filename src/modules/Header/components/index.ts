export { LocationBadge } from './LocationBadge';
export { Navigation } from './Navigation';
