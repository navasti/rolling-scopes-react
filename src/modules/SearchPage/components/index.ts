export { SearchBar } from './SearchBar';
export { Cards } from './Cards';
export { Card } from './Card';
export { Tabs } from './Tabs';
