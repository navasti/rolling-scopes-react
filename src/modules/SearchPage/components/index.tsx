export { SearchBar } from './SearchBar';
export { Card } from './Card';
