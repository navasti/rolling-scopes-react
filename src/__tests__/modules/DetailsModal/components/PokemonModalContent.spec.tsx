import { PokemonModalContent } from 'modules/DetailsModal/components';
import { screen, render } from '@testing-library/react';

describe('', () => {
  it('', () => {});
});
