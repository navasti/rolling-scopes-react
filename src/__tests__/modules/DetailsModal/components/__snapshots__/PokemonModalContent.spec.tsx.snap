// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonModalContent content should match the snapshot with passed first element of pokemonsMock as a prop 1`] = `
<div>
  <div>
    <p
      class="sc-jSMfEi fVTeHB"
    >
      <b>
        Weight
        :
      </b>
       
      85
    </p>
    <p
      class="sc-jSMfEi fVTeHB"
    >
      <b>
        Height
        :
      </b>
       
      6
    </p>
    <p
      class="sc-jSMfEi fVTeHB"
    >
      <b>
        Base experience
        :
      </b>
       
      62
    </p>
    <p
      class="sc-jSMfEi fVTeHB"
    >
      <b>
        Types:
      </b>
       
      fire
    </p>
    <div
      class="sc-hKMtZM isCUBw"
    >
      <span
        class="sc-eCYdqJ hhpgjz"
      >
        Pokemon abilities:
      </span>
      <p>
        Blaze
         - hidden: 
        no
        , slot: 
        1
      </p>
    </div>
    <div
      class="sc-hKMtZM isCUBw"
    >
      <span
        class="sc-eCYdqJ hhpgjz"
      >
        Pokemon statistics:
      </span>
      <p>
        Attack
         - power: 
        52
        , effort: 
        0
      </p>
    </div>
  </div>
</div>
`;
